### PR TITLE
Add i18n checks to PR workflows

### PR DESCRIPTION
**Pull request by AMoo-Miki**
### Description

* Add i18n checks to PR workflows
* Ignore missing `formats` while checking locale files
* Add help text and description to `i18n-check`
* Fix malformed translations

## Changelog
- infra: Add i18n checks to PR workflows
- feat: Ignore missing `formats` while checking locale files
- feat: Add help text and description to `i18n-check`
- fix: Fix malformed translations

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [X] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by ashwin-pc** [comment]
@AMoo-Miki Tests where pass?

**Comment by AMoo-Miki** [comment]
> @AMoo-Miki Tests where pass?

Tests pass when no translation files are available. I removed this ignoring in the PR that split stuff and kept them separate to not block the workflows. That PR will be the last one of i18n that we merge for everything to kick in.

**Review by manasvinibs — DISMISSED** [review]

**Review by ZilongX — DISMISSED** [review]

**Review by SuZhou-Joe — APPROVED** [review]

File: src/dev/run_i18n_check.ts [hunk 1]
```diff
@@ -121,11 +123,13 @@ run(
...
                   ignoreUnused: !!ignoreUnused,
                   ignoreMissing: !!ignoreMissing,
+                  // By default ignore missing formats
+                  ignoreMissingFormats: ignoreMissingFormats !== false,
                   fix,
                 },
                 log
               ),
-              { exitOnError: true }
+              { exitOnError: false }
```
Comment (ruanyl) [inline src/dev/run_i18n_check.ts:132]: This seems changed the existing behavior, is it by intention?

**Review by ruanyl — APPROVED** [review]

File: src/dev/run_i18n_check.ts (hunk 1 above)
Comment (AMoo-Miki) [reply src/dev/run_i18n_check.ts:132]: Yeah. If `true`, it exits at the first error. When false, it logs them all and then finally throws.

While "checking" we would rather have all of the problems shown to us, as opposed to just the first problem.

**Pull request merged**